docs/DeveloperGuide.adoc

```diff
@@ -96,64 +112,413 @@ The `UI` component,
+
+
+Step 5. The user then executes `group add BTS --i 00003` to add the correct contact as a member on the `Group`
+ and view the `Contact` profiles.
+ This calls `AddMemberToGroupCommand` which then updates the `Group` ensuring that all memberIds correspond to an existing
+ `ContactId` found in `AddressBook`. The contacts of the group are also displayed to the user.  +
+
+.State after user views contacts of Group "BTS"
+image::GroupContactsState4.png[]
+
+Here, the `GroupBook` model state is updated and memberId of 00013 from the previous step is removed while `Contact`
+ with contactId of 00003 is added into the `Group`. This change is also reflected in the `UI`.
+Now all is as it should be in `Group` "BTS". +
+
+
+The following sequence diagram summarizes what happens when a user executes a `FindGroupCommand` which
+ which updates the `Group` similar to how `AddMemberToGroupCommand` does in the above example:
+
+.Sequence diagram of executing `FindGroupCommand` to view contacts in a `Group`
+image::GroupSequenceDiagram.png[]
+
+==== Design Considerations
+
+===== Aspect: How groups stores contacts
+
+* **Alternative 1:** Stores `ContactId` class in a `Collection` in `Group`
+** Pros: Easy to get `ContactId` from `Group` to retrieve `Contact` classes from `Addressbook`.
+** Cons: Increases coupling to implementation of `Contact`. Storage and retrieval after reloading the app would also
+ cause new instances of `ContactId` to be created when loading `Group` or would require more complicated
+ loading of `Group` from storage having to happen after `AddressBook` is loaded and having to reference
+ `Contact` classes to ensure the same `ContactId` class is referenced by both `Contact` and `Group` it is in.
+* **Alternative 2 (current choice):** Stores `Collection` of Strings which are able to uniquely identify `Contact`.
+** Pros: Group classes are less coupled to implementation of `Contact`. Simpler to load `Group` classes from storage.
+ due to not needing to check and obtain a reference to `ContactId`. User input is also parsed as Strings.
+** Cons: Deciding when to check if members are still part of a `Group` since it need not be done at loading time.
+ While it is more flexible, can be a potential source of confusion as it may be possible to forget to update
+ the members in `Group`.
+
+// end::groupcontacts[]
+
+=== [Proposed] Tagging feature
+==== Description
+
+The user can tag a note with many tags by using <<UserGuide#note-tag, `note tag`>> command.
+
+==== Proposed Implementation
+
+In order to add tagging feature we will need to take a look at two processes, which are the tag command creation and the
+ execution of the command.
+
+===== Creating Tag Command
+
+We will use a TagParserUtil to create a tag from user input.
+
+Given below is an example scenario when a user tag a note with 2 tags.
+
+**Step 1:** The user command will be passed to `TaglineParser`, all the way to the `TagNoteParser`.
+
+image::CreatingTagNoteCommand.png[]
```

Comment:
<img width="916" alt="Screenshot 2019-10-25 at 2 26 30 PM" src="https://user-images.githubusercontent.com/37838247/67548089-8481c280-f733-11e9-87b3-19c0a6d6b6ee.png">
The return arrow is missing from the bottom loop. It is optional but it might be good to cinclude it here to maintain consistency across the entire diagram.